src/python/pants/goal/run_tracker.py

```diff
@@ -408,19 +410,26 @@ def run_information(self):
       run_information['target_data'] = ast.literal_eval(target_data)
     return run_information
 
+  def _stats(self):
+    if self.get_options().stats_version == 1:
+      return {
+        'run_info': self.run_information(),
+        'cumulative_timings': self.cumulative_timings.get_all(),
+        'self_timings': self.self_timings.get_all(),
+        'critical_path_timings': self.get_critical_path_timings().get_all(),
+        'artifact_cache_stats': self.artifact_cache_stats.get_all(),
+        'pantsd_stats': self.pantsd_stats.get_all(),
+        'outcomes': self.outcomes,
+        'recorded_options': self._get_options_to_record(),
+      }
+    else:
+      reporter = self.report.json_reporter
+      assert reporter, 'A JsonReporter is required for v2 stats'
+      return json.loads(open(reporter.report_path).read())
+
   def store_stats(self):
     """Store stats about this run in local and optionally remote stats dbs."""
-
-    stats = {
-      'run_info': self.run_information(),
-      'cumulative_timings': self.cumulative_timings.get_all(),
-      'self_timings': self.self_timings.get_all(),
-      'critical_path_timings': self.get_critical_path_timings().get_all(),
-      'artifact_cache_stats': self.artifact_cache_stats.get_all(),
-      'pantsd_stats': self.pantsd_stats.get_all(),
-      'outcomes': self.outcomes,
-      'recorded_options': self._get_options_to_record(),
-    }
+    stats = self._stats()
```

Comment:
The effect of this for v2 is that we load the stats from one place and store them to another?... that's a bit odd.

Reply:
✔️ I believe the latest changes help resolve this, the JSON reporter is now simply collects workunit data and serves that to the `RunTracker`.